sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -259,6 +338,99 @@ class wi_element {
                         PI_INVALID_DEVICE);
 #endif // __SYCL_DEVICE_ONLY__
   }
+  friend T operator/(const wi_element<T, NumRows, NumCols, Layout, Group> &lhs,
+                     const T &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_VectorExtractDynamic(lhs.M.spvm, lhs.idx) / rhs;
+#else
+    (void)rhs;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  wi_element &operator/=(const T &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    M.spvm = __spirv_VectorInsertDynamic(
+        M.spvm, __spirv_VectorExtractDynamic(M.spvm, idx) / rhs, idx);
+    return *this;
+#else
+    (void)rhs;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
```

Comment:
Implementation looks good, but if you feel like avoiding replicated code I believe the following should work as well. It applies to all the other similar operators as well. Just a suggestion. 😄 
```suggestion
  friend wi_element &operator/=(wi_element<T, NumRows, NumCols, Layout, Group> &lhs, const T &rhs) {
#ifdef __SYCL_DEVICE_ONLY__
    lhs = lhs / rhs;
    return lhs;
#else
    (void)lhs;
    (void)rhs;
    throw runtime_error("joint matrix is not supported on host device.",
                        PI_INVALID_DEVICE);
#endif // __SYCL_DEVICE_ONLY__
  }
```
You could even just have the `__SYCL_DEVICE_ONLY__` case and rely on `/` to fail on host, though that might make the stack trace a little confusing. I am not sure it matters in this case however.

Reply:
I think it is a nice catch, but we hope merge it soon. so, Maybe later i will address it